### PR TITLE
Use sum when templates in asr_2_3_person_cases report

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/ICDS-871

Start using the functionality added in https://github.com/dimagi/commcare-hq/pull/25880

@emord @calellowitz What's the expected due diligence in testing here?

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

File: corehq/apps/userreports/reports/sum_when_templates.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    type = TypeProperty('year_range')
+    expression = "year >= ? and year < ?"
+
+
+class UnderXMonthsTemplateSpec(SumWhenTemplateSpec):
+    type = TypeProperty("under_x_months")
+    expression = "age_at_registration < ?"
+
+
+### Templates for ICDS
```
Comment (emord) [inline corehq/apps/userreports/reports/sum_when_templates.py]: hmm. unfortunately it looks like we have a way for every other type of custom expression( https://github.com/dimagi/commcare-hq/blob/master/settings.py#L1937), but not these

**Review by emord — APPROVED** [review]
> What's the expected due diligence in testing here?

So theoretically, you could make these changes in a different report, and compare those reports on live data (using this setting: https://github.com/dimagi/commcare-hq/blob/master/settings.py#L894) from users using [laboratory](https://github.com/joealcorn/laboratory), but that's really only been used for much larger changes like changing the underlying data source (either new UCR data source, or the Citus backend).

For this change, I think its sufficient to verify that the SQL generate is the same in some way. You could also find a user on india with known values for these reports and test before/after a deploy of that branch

**Comment by orangejenny** [comment]
@emord Thanks. I found a location on india with data, will check before & after deploy.

**Comment by orangejenny** [comment]
Deployed to india, compared MPR (Delhi Oct 2019) and ASR (Andhra Pradesh), data is the same pre and post deploy.

**Pull request merged**